### PR TITLE
go: add 1.18 and 1.17.8

### PR DESCRIPTION
**Pull request by paulbry**
Adding latest [Go releases](https://go.dev/doc/devel/release)

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256.

Will defer final decision to package maintainer (i.e., @alecbcs).

**Comment by alecbcs** [comment]
Looks good to me! Thanks @paulbry for submitting this!

**Review by alecbcs — APPROVED** [review]

**Comment by tldahlgren** [comment]
@spackbot run pipeline

**Pull request merged**
